src/compas_cgal/straight_skeleton_2.py

```diff
@@ -66,3 +66,76 @@ def create_interior_straight_skeleton_with_holes(points, holes) -> PolylinesNump
         hole = np.asarray(points, dtype=np.float64)
         H.append(hole)
     return straight_skeleton_2.create_interior_straight_skeleton_with_holes(V, H)
+
+
+def create_offset_polygons_2(points, offset):
+    """Compute the polygon offset.
+
+    Parameters
+    ----------
+    points : list of point coordinates or :class:`compas.geometry.Polygon`
+        The points of the polygon.
+    offset : float
+        The offset distance. If negative, the offset is outside the polygon, otherwise inside.
+
+    Returns
+    -------
+    list of Polygon
+        The offset polygon(s).
+
+    Raises
+    ------
+    ValueError
+        If the normal of the polygon is not [0, 0, 1].
+    """
+    points = list(points)
+    if not TOL.is_allclose(normal_polygon(points, True), [0, 0, 1]):
+        raise ValueError("Please pass a polygon with a normal vector of [0, 0, 1].")
```

Comment:
perhaps something like this is more informative?

`"The normal of the polygon should be [0, 0, 1]. The normal of the provided polygon is {}".format(normal_points(points))`